coda-prototype/initUI.js

```diff
@@ -732,88 +737,22 @@ function initUI(dataset) {
-                for (let codeRow of parsedObjs) {
-
-                    let id = codeRow.hasOwnProperty("scheme_id"),
-                        name = codeRow.hasOwnProperty("scheme_name"),
-                        code_id = codeRow.hasOwnProperty("code_id"),
-                        code_value = codeRow.hasOwnProperty("code_value"),
-                        code_colour = codeRow.hasOwnProperty("code_colour"),
-                        code_shortcut = codeRow.hasOwnProperty("code_shortcut"),
-                        code_words = codeRow.hasOwnProperty("code_words"),
-                        code_regex = codeRow.hasOwnProperty("code_regex"); // TODO: Difference: This is not present in FileIO.loadCodeScheme
-
-                    if (id && name && code_id && code_value) {
-
-                        // TODO: Difference: this is not in FileIO.loadCodeScheme.
-                        // TODO: In this case, this makes sense, because this version of the function is updating
-                        // TODO: rather than adding a new code scheme. Maybe add a flag for this in loadCodeScheme?
-                        if (codeRow["scheme_id"] != tempScheme["id"]) {
-                            console.log("ERROR: Trying to upload scheme with a wrong ID");
-                            return; // todo UI error message
-                        }
 
-                        if (!newScheme) {
-                            newScheme = new CodeScheme(codeRow["scheme_id"], codeRow["scheme_name"], false);
-                        }
-
-                        let newShortcut = codeRow["code_shortcut"];
-                        if (codeRow["code_shortcut"].length === 1 && Number.isNaN(parseInt(codeRow["code_shortcut"]))) {
-                            // initialize shortcuts
-                            newShortcut = UIUtils.ascii(codeRow["code_shortcut"]);
-                        }
-
-                        // TODO: Difference: This does not exist in FileIO, but why not. This is a bug; fix.
-                        let newCode;
-                        if (code_regex && typeof codeRow["code_regex"] === "string") {
-                            newCode = new Code(newScheme, codeRow["code_id"], codeRow["code_value"], codeRow["code_colour"], newShortcut, false, [codeRow["code_regex"], "g"]);
-                        } else {
-                            newCode = new Code(newScheme, codeRow["code_id"], codeRow["code_value"], codeRow["code_colour"], newShortcut, false);
-                        }
-
-                        if (code_words) {
-                            if (codeRow["code_words"].length !== 0) {
-                                let words = codeRow["code_words"].split(",");
-                                if (words.length > 0) {
-                                    newCode.addWords(words);
-                                }
-                            }
-                        }
+        if (files.length > 0) {
+            let file = files[0];
```

Comment:
What happens if length > 1?

Reply:
One of the multi-selected files is chosen arbitrarily.

I think this is relying on the file picker being configured to only allow one file to be selected. However, I will put in an error case for this, again on io-refactoring-deduplicate-alerts.